setup-model.sh

```diff
@@ -47,3 +60,188 @@ if [ ! -e "coco.names" ]; then
 else
     echo "You already have coco.names"
 fi
+
+
+if [ ! -e "rtmdet/end2end.engine" ]; then
+    if [ $arch != "x86_64" ] && [ $arch != "aarch64" ]; then
+        echo "Unknown architecture: $arch"
+        exit 1
+    fi
+    dc_file=$dc_dir/docker-compose-people-setup-model.yaml
+    if [ $arch = "x86_64" ]; then
+        people_service=people-setup-model
+    elif [ $arch = "aarch64" ]; then
+        people_service=people-jetson-setup-model
+    fi
+
+    docker compose -f $dc_file run --rm $people_service bash -c "exit"
+    if [[ $? -ne 0 ]]; then
+        echo "build docker at first to prepare rtmdet model"
+        exit 1
+    fi
+
+    # setting for input image size
+    model_input_height=384
+    # model_input_height=416
+    # model_input_height=512
+    # model_input_height=640
+    if [ $model_input_height -eq 384 ]; then
+        model_input_width=640
+    else
+        model_input_width=$model_input_height
+    fi
+
+    # settings for RTMDet-tiny model
+    # model_url=https://download.openmmlab.com/mmdetection/v3.0/rtmdet/rtmdet_tiny_8xb32-300e_coco/rtmdet_tiny_8xb32-300e_coco_20220902_112414-78e30dcc.pth
+    # mmdet_config_filename=rtmdet_tiny_8xb32-300e_coco_reduce-postprocess.py
+
+    # settings for RTMDet-s model
+    model_url=https://download.openmmlab.com/mmdetection/v3.0/rtmdet/rtmdet_s_8xb32-300e_coco/rtmdet_s_8xb32-300e_coco_20220905_161602-387a891e.pth
+    mmdet_config_filename=rtmdet_s_8xb32-300e_coco_reduce-postprocess.py
+
+    # settings for RTMDet-m model
+    # model_url=https://download.openmmlab.com/mmdetection/v3.0/rtmdet/rtmdet_m_8xb32-300e_coco/rtmdet_m_8xb32-300e_coco_20220719_112220-229f527c.pth
+    # mmdet_config_filename=rtmdet_m_8xb32-300e_coco_reduce-postprocess.py
+
+    # deploy RTMDet model
+    mmdeploy_config_filename=detection_tensorrt-fp16_static-${model_input_height}x${model_input_width}.py
+    model_filename=$(basename $model_url)
+    if [ ! -e "rtmdet/$model_filename" ]; then
+        echo "Downloading rtmdet/$model_filename"
+        wget -P rtmdet $model_url
+    else
+        echo "You already have rtmdet/$model_filename"
+    fi
+    docker compose -f $dc_file run --rm $people_service bash -c "
```

Comment:
I will create separated script file.

Reply:
Do we need to run the script to prepare model in the production image? 
If the exported mmdeploy models are included in the production image, we can run them as YOLO.

Reply:
What is the best way to deliver model files?
We will not build images/workspaces on the robot; we will just pull images from dockerhub.
I guess that the model files will be also in the image. Are there any license issues?

Reply:
We can include model files that are created by setup-model.sh in the production image.

Because of the compatibility issue, TensorRT model files are not provided.
TensorRT models should be exported by using the same versions of libraries (CUDA, TensorRT) which are used on runtime environment.
The versions of libraries will be same in development image and production image.
So, we can copy models which are exported by setup-model.sh in the production image.

Reply:
> We can include model files that are created by setup-model.sh in the production image.

Currently, you need to build an image, then run setup-model.sh with the image. So if we want to include it in production image, we need to have multiple stage build with a script. I want to put all the process in a Dockerfile. 

The current script is in the argument of the docker compose file. So please write a shell file so that we can include the shell script to build the model in Dockerfile

Reply:
I see. I will call the shell script from Dockerfile.